src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Circle extends Figure implements AreaCalculator {
```

Comment:
Так как все классы-наследники Figure должны имплементировать интерфейс AreaCalculator, то лучше импелементировать его родительским классом Figure, тогда все наследники будут имплементировать его автоматически
```suggestion
public class Circle extends Figure {
```

Reply:
Не уверен, что всем геометрическим фигурам следует имплементировать интерфейс AreaCalculator. Например такие простейшие фигуры как точка, линия, луч, отрезок не имеют площади, если не говорить о каких-то бесконечно малых величинах.

Reply:
Это, конечно, зависит от бизнес задачи, но площадь линии или точки просто равна 0(если не влазить в бесконечно малые величины).  Давай будем считать, что у нас используются только фигуры с площадью

Reply:
Исправил в качестве решения данной бизнес задачи.

Тем не менее в защиту моей версии реализации хочу отметить, что у точки не может быть площади, поскольку у нее нет размеров, есть только положение в пространстве. Также у линии измеряют только длину. Ширины и толщины она не имеет. Из этих соображений мы неожиданно озадачим наследователей класса фигур, кому придется реализовывать метод расчета площади для таких крайних случаев как точка и линия.